detectable/src/main/java/com/synopsys/integration/detectable/factory/DetectableFactory.java

```diff
@@ -596,8 +595,7 @@ public PipInspectorDetectable createPipInspectorDetectable(
     }
 
     public PnpmLockDetectable createPnpmLockDetectable(DetectableEnvironment environment, PnpmLockOptions pnpmLockOptions) {
-        PnpmYamlTransformer pnpmYamlTransformer = new PnpmYamlTransformer(pnpmLockOptions.getDependencyTypeFilter());
-        PnpmLockYamlParser pnpmLockYamlParser = new PnpmLockYamlParser(pnpmYamlTransformer);
+        PnpmLockYamlParser pnpmLockYamlParser = new PnpmLockYamlParser(pnpmLockOptions.getDependencyTypeFilter());
```

Comment:
The parser comes into play first. It now reads in the yaml file and then decides based on its contents which transformer to create and use.